### PR TITLE
Scan project/project/... folders

### LINKED ISSUE #2548: Not updating dependencies in `/project/project/` folder
Examples:
* https://github.com/playframework/playframework/blob/main/project/project/buildinfo.sbt
* https://github.com/sbt/sbt-web-build-base/blob/master/project/project/buildinfo.sbt

scala-steward did not open a pull request on it's last run to update that library, even though there is a new version available:
https://repo1.maven.org/maven2/com/eed3si9n/sbt-buildinfo_2.12_1.0/0.11.0/

### PR DESCRIPTION
**Pull request by mkurz**
Fixes #2548

### CODE REVIEW

**Comment by mkurz** [comment]
Anyone?

File: modules/core/src/main/scala/org/scalasteward/core/buildtool/sbt/SbtAlg.scala [hunk 1]
```diff
@@ -110,7 +110,7 @@ final class SbtAlg[F[_]](config: Config)(implicit
       projectDir = buildRootDir / "project"
       files0 <- (
         fileAlg.walk(buildRootDir, 1).filter(_.extension.contains(".sbt")) ++
-          fileAlg.walk(projectDir, 1).filter(_.extension.exists(Set(".sbt", ".scala")))
+          fileAlg.walk(projectDir).filter(_.extension.exists(Set(".sbt", ".scala")))
```
Comment (exoego) [inline modules/core/src/main/scala/org/scalasteward/core/buildtool/sbt/SbtAlg.scala:113]: I think depth `2` is enough to scan `project/project`.

```suggestion
          fileAlg.walk(projectDir, 2).filter(_.extension.exists(Set(".sbt", ".scala")))
```

Comment (mkurz) [reply modules/core/src/main/scala/org/scalasteward/core/buildtool/sbt/SbtAlg.scala:113]: Believe it or not, I do have a (non public) project which has a `project/project/project/plugins.sbt` file. So I personally would prefer a depth of 3 😉 
```suggestion
          fileAlg.walk(projectDir, 3).filter(_.extension.exists(Set(".sbt", ".scala")))
```

Comment (exoego) [reply modules/core/src/main/scala/org/scalasteward/core/buildtool/sbt/SbtAlg.scala:113]: Oh... then 3 is good to go.

**Comment by mkurz** [comment]
Thanks @exoego could you also have a look at https://github.com/scala-steward-org/scala-steward/issues/2608 😉  ?

**Comment by exoego** [comment]
Sorry for long waiting and thanks your effort.

**Comment by mkurz** [comment]
@fthomas @exoego Do you know why this is not working? scala-steward nevery updated https://github.com/playframework/playframework/blob/main/project/project/buildinfo.sbt but with this patch here it actually should... Thanks for you help!

**Pull request merged**